src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -242,17 +249,24 @@ impl<'tcx> UniversalRegions<'tcx> {
         (FIRST_GLOBAL_INDEX..self.num_universals).map(RegionVid::new)
     }
 
-    /// True if `r` is classied as a global region.
+    /// True if `r` is classified as a global region.
     pub fn is_global_free_region(&self, r: RegionVid) -> bool {
         self.region_classification(r) == Some(RegionClassification::Global)
     }
 
-    /// True if `r` is classied as an external region.
+    /// True if `r` is classified as an external region.
     pub fn is_extern_free_region(&self, r: RegionVid) -> bool {
         self.region_classification(r) == Some(RegionClassification::External)
     }
 
-    /// True if `r` is classied as an local region.
+    /// True if `r` is a free region that is classified as global or
+    /// extern.  This is an important category, because these regions
+    /// can be referenced in `ClosureRegionRequirements`.
+    pub fn is_non_local_free_region(&self, r: RegionVid) -> bool {
+        self.region_classification(r) == Some(RegionClassification::Local)
```

Comment:
is there a missing `!` here?

Reply:
Yes -- but this also appears to be dead-code. I thought I remembered finding (and fixing...) this particular bug, but I can't seem to find that in my other branches now. Anyway, I will remove these bits of dead code I think.